### PR TITLE
Get pwcache from NetBSD

### PR DESCRIPTION
**Pull request by franciscozdo**
Simply copied files `pwcache.c` and `pwcache.h`.

### CODE REVIEW

**Comment by franciscozdo** [comment]
*Diffs*
`pwcache.c`
```
1,2d0
< /*    $NetBSD: pwcache.c,v 1.32 2018/11/08 06:43:52 msaitoh Exp $     */
<
73,81d70
< #if defined(LIBC_SCCS) && !defined(lint)
< #if 0
< static char sccsid[] = "@(#)cache.c   8.1 (Berkeley) 5/31/93";
< #else
< __RCSID("$NetBSD: pwcache.c,v 1.32 2018/11/08 06:43:52 msaitoh Exp $");
< #endif
< #endif /* LIBC_SCCS and not lint */
<
< #include "namespace.h"
98,103d86
< #endif
<
< #ifdef __weak_alias
< __weak_alias(user_from_uid,_user_from_uid)
< __weak_alias(group_from_gid,_group_from_gid)
< __weak_alias(pwcache_groupdb,_pwcache_groupdb)
```
`pwcache.h`
```
1,2d0
< /*    $NetBSD: pwcache.h,v 1.5 2003/11/10 08:51:51 wiz Exp $  */
<
```

File: lib/libc/gen/pwcache.c [hunk 1]
```diff
@@ -1,30 +1,592 @@
...
+
+  puts("");
+  printf("pass 3 (null pointers)\n");
+  printf("pwcache_userdb returned %d\n", pwcache_userdb(NULL, NULL, NULL));
 
-  *gid = gr->gr_gid;
-  return 0;
+  return (0);
 }
+#endif /* TEST_PWCACHE */
```
Comment (pj1031999) [inline lib/libc/gen/pwcache.c:591]: Do you plan to use that part of code?

**Review by pj1031999 — APPROVED** [review]
LGTM

File: lib/libc/gen/pwcache.c (hunk 1 above)
Comment (cahirwpz) [reply lib/libc/gen/pwcache.c]: I removed it.

**Review by cahirwpz — APPROVED** [review]
LGTM

**Pull request merged**